projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -2756,22 +2757,53 @@ function ( $selected ) {
 	}
 
 	/**
-	 * Download exported data as CSV
+	 * Admin-post handler for CSV export
 	 */
-	public function download_feedback_as_csv() {
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- verification is done on get_feedback_entries_from_post function
-		$post_data = wp_unslash( $_POST );
-		$data      = $this->get_feedback_entries_from_post();
+	public function admin_post_feedback_export() {
+		$feedback_ids_str = sanitize_text_field( wp_unslash( $_GET['feedback_ids'] ?? '' ) );
+		$post_id          = sanitize_text_field( wp_unslash( $_GET['post_id'] ?? '' ) );
+		$nonce            = sanitize_text_field( wp_unslash( $_GET['nonce'] ?? '' ) );
+
+		if ( empty( $feedback_ids_str ) || empty( $nonce ) ) {
+			wp_die( esc_html__( 'Invalid request parameters.', 'jetpack-forms' ), 400 );
+		}
+
+		$feedback_ids = explode( ',', $feedback_ids_str );
+		$feedback_ids = array_map( 'intval', $feedback_ids );
```

Comment:
May be clean up any empty values if present

```suggestion
		$feedback_ids = array_values( array_filter( array_map( 'intval', $feedback_ids ) ) );
```

Also, I suggest that we do that `'Invalid request parameters.'` check after this step, because `"0,0"` will pass that check above but won't pass if we check after this step.